cloudspanner/src/main/java/com/yahoo/ycsb/db/CloudSpannerClient.java

```diff
@@ -0,0 +1,399 @@
+     */
+    static final String NUM_CHANNELS = "cloudspanner.channels";
+  }
+
+  private static int fieldCount;
+
+  private static boolean queriesForReads;
+
+  private static int batchInserts;
+
+  private static TimestampBound timestampBound;
+
+  private static String standardQuery;
+
+  private static String standardScan;
+
+  private static final ArrayList<String> STANDARD_FIELDS = new ArrayList<>();
+
+  private static final String PRIMARY_KEY_COLUMN = "id";
+
+  private static final Logger LOGGER = Logger.getLogger(CloudSpannerClient.class.getName());
+
+  // Static lock for the class.
+  private static final Object CLASS_LOCK = new Object();
+
+  // Single Spanner client per process.
+  private static Spanner spanner = null;
+
+  // Single database client per process.
+  private static DatabaseClient dbClient = null;
+
+  // Buffered mutations on a per object/thread basis for batch inserts.
+  // Note that we have a separate CloudSpannerClient object per thread.
+  private final ArrayList<Mutation> bufferedMutations = new ArrayList<>();
+
+  private static void constructStandardQueriesAndFields(Properties properties) {
+    String table = properties.getProperty(CoreWorkload.TABLENAME_PROPERTY, CoreWorkload.TABLENAME_PROPERTY_DEFAULT);
+    standardQuery = new StringBuilder()
+        .append("SELECT * FROM ").append(table).append(" WHERE id=@key").toString();
+    standardScan = new StringBuilder()
+        .append("SELECT * FROM ").append(table).append(" WHERE id>=@startKey LIMIT @count").toString();
+    for (int i = 0; i < fieldCount; i++) {
+      STANDARD_FIELDS.add("field" + i);
+    }
+  }
+
+  private static Spanner getSpanner(Properties properties, String host, String project) {
+    if (spanner != null) {
+      return spanner;
+    }
+    String numChannels = properties.getProperty(CloudSpannerProperties.NUM_CHANNELS);
+    int numThreads = Integer.parseInt(properties.getProperty(Client.THREAD_COUNT_PROPERTY, "1"));
+    SpannerOptions.Builder optionsBuilder = SpannerOptions.newBuilder()
+        .setSessionPoolOption(SessionPoolOptions.newBuilder()
+            .setMinSessions(numThreads)
+            // Since we have no read-write transactions, we can set the write session fraction to 0.
+            .setWriteSessionsFraction(0)
+            .build());
+    if (host != null) {
+      optionsBuilder.setHost(host);
+    }
+    if (project != null) {
+      optionsBuilder.setProjectId(project);
+    }
+    if (numChannels != null) {
+      optionsBuilder.setNumChannels(Integer.parseInt(numChannels));
+    }
+    spanner = optionsBuilder.build().getService();
+    Runtime.getRuntime().addShutdownHook(new Thread("spannerShutdown") {
+        @Override
+        public void run() {
+          System.out.println("Closing Cloud Spanner client...");
+          spanner.closeAsync();
+          System.out.println("...done.");
```

Comment:
I removed these as they are not really necessary at all. FWIW, when using the logger, the message are not actually realized on the screen (the program exits before that has a chance to happen).